app/controllers/application_controller.rb

```diff
@@ -63,4 +63,10 @@ def prompt_to_set_up_2fa
   def prompt_to_enter_otp
     redirect_to user_two_factor_authentication_url
   end
+
+  def prompt_to_confirm_mobile(mobile, mobile_taken = false)
+    user_session[:unconfirmed_mobile] = mobile
+    user_session[:unconfirmed_mobile_taken] = mobile_taken
+    redirect_to phone_confirmation_new_path
+  end
```

Comment:
After going through the code some more, I realized there's a better way to do this that doesn't require the `mobile_taken` parameter, by using a controller concern for generating and sending the confirmation code, like this:
```ruby
module OtpGenerator
  def send_new_confirmation_code
    set_confirmation_code(generate_confirmation_code)
    SmsSenderOtpJob.perform_later(confirmation_code, unconfirmed_mobile)
  end

  private

  def set_confirmation_code(code)
    user_session[:phone_confirmation_code] = code
  end

  def generate_confirmation_code
    digits = Devise.direct_otp_length
    random_base10(digits)
  end

  def random_base10(digits)
    SecureRandom.random_number(10**digits).to_s.rjust(digits, '0')
  end

  def confirmation_code
    user_session[:phone_confirmation_code]
  end

  def unconfirmed_mobile
    user_session[:unconfirmed_mobile]
  end
end
```
and then including the concern in `RegistrationsController` and `ApplicationController` and updating the `process_updates` method to so that it passes in the form to ` prompt_to_confirm_mobile`:
```ruby
def process_updates(resource)
  updater = UserFlashUpdater.new(@update_user_profile_form, flash)

  updater.set_flash_message

  if @update_user_profile_form.mobile_changed?
    prompt_to_confirm_mobile(@update_user_profile_form)
  elsif is_flashing_format?
    EmailNotifier.new(resource).send_password_changed_email
    redirect_to edit_user_registration_url
  end
end
```
The new `prompt_to_confirm_mobile` method now takes a form as a parameter, like this:
```ruby
def prompt_to_confirm_mobile(form)
  user_session[:unconfirmed_mobile] = form.mobile

  send_new_confirmation_code unless form.mobile_taken?

  redirect_to phone_confirmation_path
end
```
This allows us to check if the mobile is taken by using the existing method on the form, rather than having to set a session variable and pass it around from one place to another.

To get the tests to pass with this change requires the following:
- Update `two_factor_authentication_setup_controller` so that it passes the form to `prompt_to_confirm_mobile`
- Replace `expect(response).to redirect_to(phone_confirmation_new_path)` with `expect(response).to redirect_to(phone_confirmation_path)`
- Replace `expect(Users::PhoneConfirmationController).` with `allow_any_instance_of(OtpGenerator).`